lib/util/merge-confidence/index.ts

```diff
@@ -28,41 +28,28 @@ export const confidenceLevels: Record<MergeConfidence, number> = {
 export function initConfig(): void {
   apiBaseUrl = getApiBaseUrl();
   token = getApiToken();
-  supportedDatasources =
-    parseSupportedDatasourceString() ?? presetSupportedDatasources;
+  supportedDatasources = parseSupportedDatasourceString();
```

Comment:
it seems `presetSupportedDatasources` is now obsolete? because the new config option always set's a value.

Reply:
@nabeelsaabna

Reply:
Then should I move the list to `config/constants` and import it from there where needed?

Reply:
sure

Reply:
`presetSupportedDatasources` it's still needed when the user sets a invalid value.
Since this is an global config option, we will only warn the user about this config error and continue the run.
At that time `GlobalConfig.get('mergeConfidenceDatasources')` will return the invalid value, and we need to fallback to the default value, for that `presetSupportedDatasources` will come in handy.

ie. it will be used in the `parseDatasources` fn.

Reply:
`mergeConfidenceDatasources` is option config for development /testing envs 
Any thing needed from my side after the last comment from Rahul ?

Reply:
yes. you need to be able to add new supported data sources on test systems. so we only use the current values for Schema but allow new values?

Reply:
> it seems `presetSupportedDatasources` is now obsolete? because the new config option always set's a value.

i assume you mean `presetSupportedDatasources` is not needed locally in this file ? because the config option should use it as the default value (once the [import](https://github.com/renovatebot/renovate/pull/28880#discussion_r1606526139) is fixed).

Reply:
It will still be needed in the local-file, as fallback incase the option is set to an invalid value ie. validation use-case.
Example:
https://github.com/RahulGautamSingh/renovate/blob/03c5cac905734466c9c53a8fd521183e87e0a815/lib/util/merge-confidence/index.spec.ts#L457-L460

Reply:
I don't mind a failure if we (Mend) assign invalid value in testing env
(in case it makes the code more simple)